src/snowflake/snowpark/_internal/utils.py

```diff
@@ -703,3 +703,72 @@ def strip_double_quotes_in_like_statement_in_table_name(table_name: str) -> str:
 
     # if table_name == '"a.b"', then we should call SHOW TABLES LIKE 'a.b'
     return table_name[1:-1] if table_name[0] == table_name[-1] == '"' else table_name
+
+
+def parse_table_name(table_name: str) -> List[str]:
+    """
+    This function implements the algorithm to parse a table name.
+
+    We parse the table name according to the following rule:
+        https://docs.snowflake.com/en/sql-reference/identifiers-syntax
+        Unquoted object identifiers:
+        Start with a letter (A-Z, a-z) or an underscore (“_”).
+        Contain only letters, underscores, decimal digits (0-9), and dollar signs (“$”).
+        Are stored and resolved as uppercase characters (e.g. id is stored and resolved as ID).
+        If you put double quotes around an identifier (e.g. “My identifier with blanks and punctuation.”),
+        the following rules apply: The case of the identifier is preserved when storing and resolving
+        the identifier (e.g. "id" is stored and resolved as id).
+        The identifier can contain and start with ASCII, extended ASCII, and non-ASCII characters.
+        To use the double quote character inside a quoted identifier, use two quotes. For example:
```

Comment:
It feels like an example is missing?

Reply:
yeah, I should remove this line. it's a copy-paste from our website.

Reply:
updated